### PR TITLE
Revert "Inline resource strings in the compiler (#80896)"

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
This reverts commit 759fecb5ef5ca893d289da7b885cf7f8dd50cb84.

Reverting it to fix the fallout in extra-platforms for now: https://github.com/dotnet/runtime/pull/81409#issuecomment-1411230969

There are two problems that I can see:
* Some tests are looking for the resource stream. This is not a customer scenario and not worried about it. We can just make it so that it's obvious to the compiler that the optimization needs to be disabled (e.g. add `Type.GetType("System.SR, blah").GetMethod("GetResourceString");` to the test - this will trick the compiler into thinking the optimization has been defeated).
* Event source looks like doesn't work. This is a bigger issue. I wonder if this also doesn't work with `<UseSystemResourceKeys>true</UseSystemResourceKeys>` specified. Will need to look into that more.

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

**Comment by MichalStrehovsky** [comment]
Let's not merge this if #81463 and #81466 can get in instead. Leaving open just in case there's pushback.

**Pull request closed without merging**
